src/test/java/org/folio/rest/api/ItemStorageTest.java

```diff
@@ -2373,6 +2376,192 @@ public void canSearchByPurchaseOrderLineIdentifierProperty() throws Exception {
     assertThat(poli1Items.get(0).getId(), is(firstItem.getId()));
   }
 
+  @Test
+  public void canCreateItemWithStatisticalCodeId() throws Exception {
+    final var statisticalCode = statisticalCodeFixture
+      .createSerialManagementCode(new StatisticalCodeBuilder()
+        .withCode("stcone")
+        .withName("Statistical code 1"));
+
+    final UUID holdingsRecordId = createInstanceAndHolding(mainLibraryLocationId);
+
+    final UUID statisticalCodeId = UUID.fromString(
+      statisticalCode.getJson().getString("id")
+    );
+
+    final String status = "Available";
+
+    final JsonObject itemToCreate = new ItemRequestBuilder()
+      .forHolding(holdingsRecordId)
+      .withMaterialType(journalMaterialTypeId)
+      .withPermanentLoanType(canCirculateLoanTypeId)
+      .withStatus(status)
+      .withStatisticalCodeIds(Arrays.asList(statisticalCodeId))
+      .create();
+
+    final Response createdItem = itemsClient.attemptToCreate(itemToCreate);
+
+    assertThat(createdItem.getStatusCode(), is(HttpURLConnection.HTTP_CREATED));
+  }
```

Comment:
Does this need to be a separate test? Could it be added to the general test for creating an item?

Reply:
I do not see a general can create item method to move this to. What would you suggest?

Reply:
I suggest this one - https://github.com/folio-org/mod-inventory-storage/blob/a3eb856b8006d3e2246f142b0f271539c760cfde/src/test/java/org/folio/rest/api/ItemStorageTest.java#L202

Reply:
Done.

Reply:
Moving to that method causes other tests to fail! 

```
Failed tests: 

  ItemEffectiveCallNumberComponentsTest.canCalculateEffectiveCallNumberPropertyOnUpdate:261 

Expected: is <{"id":"b4e32eee-0b16-4fa9-997c-d52aa7bb3789","_version":2,"hrid":"it00000000036","holdingsRecordId":"aa5e032e-0917-4e4f-b7fa-34215da30d8f","formerIds":[],"effectiveShelvingOrder":"","itemLevelCallNumber":"initItCN","effectiveCallNumberComponents":{"callNumber":"initItCN"},"yearCaption":[],"notes":[],"circulationNotes":[],"status":{"name":"Available","date":"2021-11-16T15:35:51.035+00:00"},"materialTypeId":"c052063c-73ee-4692-8f32-ac412cdb3793","permanentLoanTypeId":"77d8bc76-46b3-441b-813b-b47920ca57f0","temporaryLocationId":"b2c78c43-f4ab-4fe5-9fc5-76f79a79cadd","effectiveLocationId":"b2c78c43-f4ab-4fe5-9fc5-76f79a79cadd","electronicAccess":[],"statisticalCodeIds":[],"metadata":{"createdDate":"2021-11-16T15:35:51.035+00:00","updatedDate":"2021-11-16T15:35:51.035+00:00"},"instanceId":"2eaa2452-5ee0-4ac9-97f8-73e7bd698eea"}>

     but: was <{"instanceId":"2eaa2452-5ee0-4ac9-97f8-73e7bd698eea","id":"b4e32eee-0b16-4fa9-997c-d52aa7bb3789","_version":1,"hrid":"it00000000036","holdingsRecordId":"aa5e032e-0917-4e4f-b7fa-34215da30d8f","formerIds":[],"effectiveShelvingOrder":"","itemLevelCallNumber":"initItCN","effectiveCallNumberComponents":{"callNumber":"initItCN"},"yearCaption":[],"notes":[],"circulationNotes":[],"status":{"name":"Available","date":"2021-11-16T15:35:51.035+00:00"},"materialTypeId":"c052063c-73ee-4692-8f32-ac412cdb3793","permanentLoanTypeId":"77d8bc76-46b3-441b-813b-b47920ca57f0","temporaryLocationId":"b2c78c43-f4ab-4fe5-9fc5-76f79a79cadd","effectiveLocationId":"b2c78c43-f4ab-4fe5-9fc5-76f79a79cadd","electronicAccess":[],"statisticalCodeIds":[],"metadata":{"createdDate":"2021-11-16T15:35:51.035+00:00","updatedDate":"2021-11-16T15:35:51.035+00:00"}}>
```

Reply:
I moved that requested change to another branch, https://github.com/folio-org/mod-inventory-storage/tree/MODINVSTOR-676-failure. There appears to be that tests are not independent.

Reply:
You might find those are unrelated / false failures.

Those tests are very strict on the output check (they check the whole JSON), often the only difference is the `_version` property.

It is useful that it reveals something strange about how that property is incremented, however it makes the suite generally unstable.

Reply:
Well, now failing on any change. Something wrong with the tests. Going to pull this request change back in.

Reply:
Okay, it passed this time. I think there needs to be an issue to resolve the sporadic failing tests. Certainly beyond scope of this issue.

Reply:
By all means raise the issue (one may already exist). As is often the case, these issues tend not to get prioritised.